### PR TITLE
Documentation for MailboxMailing

### LINKED ISSUE #2: Provide documentation
1.0 should come with a complete documentation in MKDocs format.

systopia-reference: 10433

### PR DESCRIPTION
**Pull request by EnnoBelle**
*systopia-reference: 26334*

### CODE REVIEW

**Comment by bjendres** [comment]
Thanks @EnnoBelle

File: docs/Documentation/Configuration.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+## Configuration
```
Comment (jensschuppe) [inline docs/Documentation/Configuration.md:1]: The general title of a page should be a H1 level heading, all other headings should be adjusted accordingly.

File: docs/Documentation/Configuration.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+Before you start with the configuration process, please ensure that the extension is properly installed on a working CiviCRM and that you have access to a functioning mailbox that you can and want to use for your MailboxMailing setup.
+
+### Compatability
```
Comment (jensschuppe) [inline docs/Documentation/Configuration.md:5]: `s/Compatability/Compatibility`

File: docs/Documentation/Configuration.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
+## Configuration
+
+Before you start with the configuration process, please ensure that the extension is properly installed on a working CiviCRM and that you have access to a functioning mailbox that you can and want to use for your MailboxMailing setup.
```
Comment (jensschuppe) [inline docs/Documentation/Configuration.md:3]: The "official" extension name is *Mailbox Mailing* (with a space) - applies to other occurrences as well.

File: docs/Documentation/Configuration.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+### Compatability
+
+The MailboxMailing extension is compatible with every IMAP or POP3 mailserver. You need access to the mailbox via webmail or you can use basically every modern mailing program such as Gmail, Outlook, Apple icloud mail, yahoo and many others. You can also use Maildir oder Localdir to use the extension without having to connect through a mailserver like Outlook or Gmail.
+
+### Set up the Mailbox Connection
+
+To set up the Mailbox Connection go to *Administer* - *Administration Console* - *MailboxMailing settings.* Here you can *add Mailbox settings.* You can set a names for the groups of settings and create multiple groups for different mailboxes.
```
Comment (jensschuppe) [inline docs/Documentation/Configuration.md:11]: Let's find another term for what is described as "groups of settings", as "Group" is a specific CiviCRM entity that is used in another context and might be confusing.

File: docs/Documentation/Configuration.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+- **Source:** Folder or path to poll from. For IMAP/POP3, this is typically "INBOX." For Maildir or Localdir, this is the directory path.
+    
+- **Use SSL?:** Whether to use SSL (Secure Sockets Layer) for IMAP and POP3 or not. SSL ensures that the connection to the mail server is encrypted and secure. For Maildir and Localdir, there is no need for SSL settings, since they establish no network connection.
+    
+- **Sender Group:** The group of contacts allowed to send mailings using these MailSettings. Only emails from contacts in this group will be processed.
+    
+- **Recipient Group:** The group of contacts to receive mailings using these MailSettings. Only contacts in this group will be sent mailings.
+    
+
+The **Sender Group** and **Recipient Group** should consist of CiviCRM contact groups. This feature ensures that only authorized individuals can send emails and that specific groups receive the messages. This allows for effective management of incoming and outgoing emails.
```
Comment (jensschuppe) [inline docs/Documentation/Configuration.md:34]: > should consist of CiviCRM contact groups

Actually, they *are* groups and need to be created before in order to be available in those settings fields.

File: docs/Documentation/Configuration.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+- **Subject:** A pattern to use as the subject for the Mailing. Here you can use Tokens to automatically the required data.
```
Comment (jensschuppe) [inline docs/Documentation/Configuration.md:36]: > Here you can use Tokens to automatically the required data.

This sentence no verb :wink:

File: docs/Documentation/Configuration.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    
+- **Notify sender about errors:** Whether to notify senders about errors.
+    
+- **Error notification subject:** The subject to use for notifications about errors sent to senders.
+    
+- **Template for error notifications of sender:** A Smarty template to use for notifications about errors sent to senders.
+    
+- **Notification Activity Type:** The activity type to use for activities to notify senders about errors.
+    
+- **Archive Mailing:** Whether to archive mailings after sending. This feature is useful for preserving records of communications. This ensures that a copy of sent emails is available for future reference.
```
Comment (jensschuppe) [inline docs/Documentation/Configuration.md:54]: Actually, if this option is disabled, mailings will be kept in the "Sent" status. Archiving moves them into the "Archived" list instead. This is not about "keep a copy".

**Review by jensschuppe — COMMENTED** [review]
General hints:
* There is no need for the `docs/Documentation` sub-directory, let's just move all files into `docs`
* Let's use *snake case* formatted file names, like `example-hints.md`
* I noticed some minor spelling issues, please have a second read
* The `mkdocs.yml` is not part of this PR, it should be adjusted accordingly, of course

**Pull request merged**